src/extent.c

```diff
@@ -483,7 +459,16 @@ extents_first_fit_locked(tsdn_t *tsdn, arena_t *arena, extents_t *extents,
 		assert(!extent_heap_empty(&extents->heaps[i]));
 		extent_t *extent = extent_heap_first(&extents->heaps[i]);
 		assert(extent_size_get(extent) >= size);
-		if (ret == NULL || extent_snad_comp(extent, ret) < 0) {
+                bool size_ok = true;
+		/*
+		 * In order to reduce fragmentation, avoid reusing and splitting
+		 * large extents for much smaller sizes.
+		 */
+		if ((sz_pind2sz(i) >> opt_lg_extent_max_active_fit) > size) {
```

Comment:
The intention was to only apply this to dirty_extents, maybe add a new arg to `extents_fit_locked` and specify if max_active_fit should be checked?

Reply:
Sure, update with restriction to delay_coalesce (which is only set for dirty).